### DISCUSSION TITLE
How to add background Opacity to component in plugin

### DISCUSSION THREAD

**Discussion by jd-solanki** [tailwindlabs/tailwindcss#5262] [Help] [answered]
Hi I am trying to add background opacity to the button component using the plugin.

```js
backgroundOpacity: theme('backgroundOpacity.50'),
```

Above code results in `background-opacity: 0.5` which is invalid CSS.

How can I replicate `bg-opacity-50` class in the plugin?

Regards.

**Comment by simonswiss** [comment, marked as answer]
Hey!

`background-opacity` is not a thing in CSS, and the way Tailwind CSS handles this is by keeping track of a `--tw-bg-opacity` CSS variable, and composing this with the `background-color` (using the opacity in the alpha channel of an RGBA color value).

Here's an example with an element where the following classes are applied:

`bg-purple-500 bg-opacity-50`

![image](https://user-images.githubusercontent.com/485747/130557860-969fc73b-0c7e-46c2-a767-1ed68a9adfc3.png)

You can see that the `bg-purple-500` sets the `--tw-bg-opacity` to be `1`, or full opacity. It then uses that variable in the background color definition.

As a result, the color will be at full opacity, unless a `bg-opacity-{n}` class is also added to the same element. In that case, when applying `bg-opacity-50` to that element, the value of `--tw-bg-opacity` is updated to be `0.5`, and therefore the background color of the element becomes a composed `purple-500` at `50%` opacity.

![image](https://user-images.githubusercontent.com/485747/130558087-ad88ec55-0b1d-4cb5-b547-31ba95c6423f.png)

So – to replicate what the `bg-opacity-50` class does in a plugin, you'd just need to redefine the `--tw-bg-opacity` variable to be `0.5` 👍

**Reply by jd-solanki** [reply]
Can you provide an example code snippet?

Right now I am converting HEX to HEXA for getting opacity.

**Reply by DanielBailey-web** [reply]
@simonswiss that helped me SOOOO much, I was trying to figure out a way to transition bg-opacity when all I needed to do was use transition-colors.... thank you so much
